### PR TITLE
PERF: Eliminate slow check for pypy during numpy import

### PR DESCRIPTION
**Pull request by eendebakpt**
This PR replaces the import of `platform` and a  call to `platform.python_implementation()` with a check using `sys.implementation.name` to improve the numpy import time.  The improvement is about 5-10 ms (system dependent).

Also see #22061 

<details><summary>Benchmark</summary>

```
python -c "import time; t0=time.perf_counter(); import sys; 'PyPy' in sys.version; dt=time.perf_counter()-t0; print(1e3*dt)"
```
Result: < 1 ms
```
python -c "import time; t0=time.perf_counter(); import platform; platform.python_implementation()=='PyPy'; dt=time.perf_counter()-t0; print(1e3*dt)"
```
Result: 10 ms

</details>

### CODE REVIEW

File: numpy/core/_internal.py [hunk 1]
```diff
@@ -16,7 +15,7 @@
 except ImportError:
     ctypes = None
 
-IS_PYPY = platform.python_implementation() == 'PyPy'
+IS_PYPY = 'PyPy' in sys.version
```
Comment (mattip) [inline numpy/core/_internal.py:18]: [PEP 421](https://peps.python.org/pep-0421/#required-attributes) added `sys.implementation` for python3 
```suggestion
IS_PYPY = sys.implementation.name == 'pypy'
```

Comment (eendebakpt) [reply numpy/core/_internal.py]: @mattip Thanks! I didn't know about that one. Do you want me to also change the implementation of`numpy.testing.IS_PYPY` with `sys.implementation.name == 'pypy'`?
(and then remove the test as well?)

Comment (mattip) [reply numpy/core/_internal.py]: > Do you want me to also change the implementation ...

Makes sense, sure.

File: numpy/testing/_private/utils.py [hunk 2]
```diff
@@ -48,7 +48,7 @@ class KnownFailureException(Exception):
 KnownFailureTest = KnownFailureException  # backwards compat
 verbose = 0
 
-IS_PYPY = platform.python_implementation() == 'PyPy'
+IS_PYPY = sys.implementation.name =='PyPY'
```
Comment (mattip) [inline numpy/testing/_private/utils.py:51]:
```suggestion
IS_PYPY = sys.implementation.name == 'PyPY'
```

File: numpy/testing/_private/utils.py [hunk 3]
```diff
@@ -48,7 +48,7 @@ class KnownFailureException(Exception):
 KnownFailureTest = KnownFailureException  # backwards compat
 verbose = 0
 
-IS_PYPY = platform.python_implementation() == 'PyPy'
+IS_PYPY = sys.implementation.name == 'PyPY'
```
Comment (mattip) [inline numpy/testing/_private/utils.py:51]:
```suggestion
IS_PYPY = sys.implementation.name == 'pypy'
```

**Comment by mattip** [comment]
I made a bit of a mess trying to do commits via the github UI, but I think it is OK now. Squash merging is desirable

**Comment by seberg** [comment]
Thanks @eendebakpt and Matti.

**Pull request merged**